libs/pageserver_api/src/key.rs

```diff
@@ -24,6 +25,60 @@ pub struct Key {
 pub const KEY_SIZE: usize = 18;
 
 impl Key {
+    /// Transform a raw key to a storage key
+    pub fn from_raw_key_fixed(key: &[u8; RAW_KEY_SIZE]) -> Self {
```

Comment:
The "raw" naming is a bit confusing -- it's not obvious what's more raw about these keys (or more "cooked" about other keys ;-))

Maybe we should think of the top level prefix as "metadata" -- it's keys which are internally generated to store metadata, rather than the keys that map directly to user data.  So it would be METADATA_KEY_PREFIX and then everything within that prefix has the properties of metadata (i.e. it is expected to be small, does not scale O(N) with user database size, and can reasonably be pinned on local disk)